controller/resource_roles.go

```diff
@@ -69,6 +82,62 @@ func (c *ResourceRolesController) ListAssignedByRoleName(ctx *app.ListAssignedBy
 	})
 }
 
+// AssignRole assigns a specific role for a resource, to one or more identities.
+func (c *ResourceRolesController) AssignRole(ctx *app.AssignRoleResourceRolesContext) error {
+
+	currentUser, err := login.ContextIdentity(ctx)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError(err.Error()))
+	}
+	if currentUser == nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError("identity ID not found in token"))
+	}
+
+	// check if the current user token belongs to a user who has the necessary privileges
+	// for assigning roles to other users.
+	hasScope, err := c.permissionService.HasScope(ctx, *currentUser, ctx.ResourceID, ROLE_ASSIGNMENT_SCOPE)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, err)
+	}
+	if !hasScope {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewForbiddenError("user is not authorized to assign roles"))
+	}
+
+	// In a batch assignment of roles, all users need to be a part of the resource
+	// Only then futher role assignments would be allowed.
+	for _, identity := range ctx.Payload.Data {
+		identityIDAsUUID, err := uuid.FromString(identity.ID)
+		assignedRoles, err := c.roleManagementService.ListAssignmentsByIdentityAndResource(ctx, ctx.ResourceID, identityIDAsUUID)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+		if len(assignedRoles) == 0 {
+			log.Error(ctx, map[string]interface{}{
+				"resource_id": ctx.ResourceID,
+				"identity_id": identityIDAsUUID,
+				"role":        ctx.RoleName,
+			}, "identity not part of a  resource cannot be assigned a role")
+			return jsonapi.JSONErrorResponse(ctx, errors.NewForbiddenError(fmt.Sprintf("identity %s does not belong to the resource", identity.ID)))
+		}
+	}
+
+	// Now that we have confirmed that all the users were part of the resource ( ex. space )
```

Comment:
Same rewording here, I suggest "Now that we have confirmed that all users have pre-existing roles"